homeassistant/components/renault/__init__.py

```diff
@@ -0,0 +1,45 @@
+"""Support for Renault devices."""
+import aiohttp
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_LOCALE, DOMAIN, PLATFORMS
+from .renault_hub import RenaultHub
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Load a config entry."""
+    renault_hub = RenaultHub(hass, config_entry.data[CONF_LOCALE])
+    try:
+        login_success = await renault_hub.attempt_login(
+            config_entry.data[CONF_USERNAME], config_entry.data[CONF_PASSWORD]
+        )
+    except aiohttp.ClientConnectionError as exc:
+        raise ConfigEntryNotReady() from exc
+
+    if not login_success:
+        return False
```

Comment:
Probably more fitting for a future PR; but we could trigger a reauth flow on this location.

Reply:
Noted.